Pathogens/Utility/location_utils/lat_lng_generation_report.py

```diff
@@ -0,0 +1,314 @@
+from enum import Enum
+from io import TextIOWrapper
+
+import re
+import Pathogens.Utility.location_utils.mapbox_api_client as mapbox_api_client
+from Pathogens.Utility.location_utils.coordinate_helpers import is_point_in_bounding_box
+from Pathogens.Utility.location_utils.country_codes import get_country_code
+
+mapbox_api_query_url_regex = re.compile(r'^https:\/\/api.mapbox.com\/geocoding\/v5\/mapbox.places\/(.+).json\?access_token=([^&]+)&types=([^&]+)&country=([^&]+)$')
+mapbox_api_query_access_token_regex = re.compile(r'access_token=([^&]+)')
+
+class MapboxApiRequestLogLevel(Enum):
+    INFO = 1,
+    WARN = 2,
+    ERROR = 3
+
+def redact_mapbox_api_key_from_mapbox_api_query_url(query_url: str):
+    if(mapbox_api_query_url_regex.search(query_url)):
+        return re.sub(mapbox_api_query_access_token_regex, 'access_token=[REDACTED]', query_url)
+
+    return "ENCOUNTERED ERROR WHEN REDACTING MAPBOX REQUEST URL, URL NOT INCLUDED"
+
+def generate_mapbox_api_request_log_prefix(
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    logLevel: MapboxApiRequestLogLevel
+):
+    string_log_level_printout_map = {
+        MapboxApiRequestLogLevel.INFO: "[INFO]",
+        MapboxApiRequestLogLevel.WARN: "[WARN]",
+        MapboxApiRequestLogLevel.ERROR: "[ERROR]",
+    }
+    city_state_country_printout = "[" + (city_name if city_name is not None else 'N/A') + ", " + \
+        (state_name if state_name is not None else 'N/A') + ", " + country_name + "]"
+
+
+    return string_log_level_printout_map[logLevel] + " - " + city_state_country_printout
+
+def format_mapbox_text_and_matching_text_for_display(mapbox_response):
```

Comment:
Q: Sean what's all this `text` + `matching_text` garbage about?

A: Yeah so basically `text` can occasionally be in the native language of that country, the big one is that most of the ones in Egypt and Sudan seem to be in Arabic. What I found was that in situations like those `matching_text` is the English localized version but `matching_text` isn't always there so I'm just displaying both and the human that actually reads the report won't get stuck with just a name in Arabic or something and will always have as much information as possible to make their decision about whether or not this location is broken. I think this is okay, one of them was always localized to English from my experience.

There's a couple examples in that text file I linked.